### PR TITLE
add AWS prefix to S3 type, Set resource type in businessObject.resources

### PR DESCRIPTION
**Pull request by dexamundsen**
- Rename Type.S3_STORAGE_FOLDER to Type.AWS_S3_STORAGE_FOLDER  (in line with WSM definition)
- Minor comment and doc updates
- Set resource type in businessObject.resources
- Upgrade WSM client

### CODE REVIEW

File: src/main/java/bio/terra/cli/service/WorkspaceManagerServiceAws.java [hunk 1]
```diff
@@ -50,6 +50,56 @@ public static WorkspaceManagerServiceAws fromContext() {
...
+        new DeleteControlledAwsResourceRequestBody().jobControl(new JobControl().id(asyncJobId));
+
+    handleClientExceptions(
+        () -> {
+          ControlledAwsResourceApi controlledAwsResourceApi =
+              new ControlledAwsResourceApi(apiClient);
+          // make the initial delete request
+          HttpUtils.callWithRetries(
+              () ->
+                  controlledAwsResourceApi.deleteAwsS3StorageFolder(
```
Comment (rogerwangcs) [inline src/main/java/bio/terra/cli/service/WorkspaceManagerServiceAws.java:65]: Should this be checking resource type before calling this? This common method will only work with s3 storage folders as is right? Or are we adding cases for other resources as we add them?

Comment (dexamundsen) [reply src/main/java/bio/terra/cli/service/WorkspaceManagerServiceAws.java:65]: It will be used with sagemaker notebooks as well part of https://github.com/DataBiosphere/terra-cli/pull/474

Comment (dexamundsen) [reply src/main/java/bio/terra/cli/service/WorkspaceManagerServiceAws.java]: Reverted this change. 
initial plan was to use a switch case and compare the resource type (similar to implementation in WSM). 
removed it since it added additional look ups and provided minimal benefit

**Review by rogerwangcs — APPROVED** [review]

**Review by zloery — APPROVED** [review]

File: src/main/java/bio/terra/cli/businessobject/resource/GcsBucket.java [hunk 2]
```diff
@@ -30,6 +30,7 @@ public class GcsBucket extends Resource {
   /** Deserialize an instance of the disk format to the internal object. */
   public GcsBucket(PDGcsBucket configFromDisk) {
     super(configFromDisk);
+    this.resourceType = Type.GCS_BUCKET;
```
Comment (yuhuyoyo) [inline src/main/java/bio/terra/cli/businessobject/resource/GcsBucket.java:33]: is this change related?

Comment (dexamundsen) [reply src/main/java/bio/terra/cli/businessobject/resource/GcsBucket.java:33]: This was added to the description, not the title since the title was already long earlier. Adding it now.

**Comment by dexamundsen** [comment]
Errors in PR tests are related to WSM issues. All tests are connected, hence a successful test requires stable WSM env. WSM fixes are merged, but not deployed to test env. Hence force merging this, needed for subsequent PR
